### PR TITLE
PP-4059 - Remove transaction model

### PR DESCRIPTION
**Pull request by kbottla**
Removed transaction model and related classes.

This change will remove dual writing of charges/refunds data to two data models and reverts back to original design i.e., write to separate charges & refunds tables only.   

with @heathd, @mrlumbu, @georgeracu  - mob programming

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java [hunk 1]
```diff
@@ -1024,6 +1024,7 @@ private String addChargeAndCardDetails(Long chargeId, ChargeStatus status, Servi
         app.getDatabaseTestHelper().addToken(chargeId, "tokenId");
         app.getDatabaseTestHelper().addEvent(chargeId, chargeStatus.getValue());
         app.getDatabaseTestHelper().updateChargeCardDetails(chargeId, cardBrand, "1234", "Mr. McPayment", "03/18", "line1", null, "postcode", "city", null, "country");
+        //TODO
```
Comment (georgeracu) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java]: What do we want to do here? :)

File: src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java [hunk 2]
```diff
@@ -10,26 +10,13 @@
 import org.mockito.runners.MockitoJUnitRunner;
 import uk.gov.pay.connector.app.ConnectorConfiguration;
 import uk.gov.pay.connector.app.LinksConfig;
-import uk.gov.pay.connector.dao.CardTypeDao;
-import uk.gov.pay.connector.dao.ChargeDao;
-import uk.gov.pay.connector.dao.ChargeEventDao;
-import uk.gov.pay.connector.dao.GatewayAccountDao;
-import uk.gov.pay.connector.dao.PaymentRequestDao;
-import uk.gov.pay.connector.dao.TokenDao;
+import uk.gov.pay.connector.dao.*;
```
Comment (georgeracu) [inline src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java]: We do not use wildcard imports :-)

File: src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java [hunk 3]
```diff
@@ -10,26 +10,13 @@
...
-import uk.gov.pay.connector.model.domain.ChargeEntity;
-import uk.gov.pay.connector.model.domain.ChargeEntityFixture;
-import uk.gov.pay.connector.model.domain.ChargeStatus;
-import uk.gov.pay.connector.model.domain.GatewayAccountEntity;
-import uk.gov.pay.connector.model.domain.PaymentRequestEntity;
-import uk.gov.pay.connector.model.domain.TokenEntity;
-import uk.gov.pay.connector.model.domain.transaction.ChargeTransactionEntity;
-import uk.gov.pay.connector.model.domain.transaction.TransactionEntity;
-import uk.gov.pay.connector.model.domain.transaction.TransactionOperation;
+import uk.gov.pay.connector.model.domain.*;
```
Comment (georgeracu) [inline src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java:23]: We do not use wildcard imports :-)

File: src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java [hunk 4]
```diff
@@ -52,18 +39,12 @@
 import static org.hamcrest.core.IsNull.nullValue;
 import static org.junit.Assert.assertThat;
 import static org.mockito.ArgumentCaptor.forClass;
-import static org.mockito.Mockito.any;
-import static org.mockito.Mockito.doAnswer;
-import static org.mockito.Mockito.never;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (georgeracu) [inline src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java]: We do not use wildcard imports :-)

File: src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java [hunk 5]
```diff
@@ -52,18 +39,12 @@
...
 import static uk.gov.pay.connector.model.ChargeResponse.ChargeResponseBuilder;
 import static uk.gov.pay.connector.model.ChargeResponse.aChargeResponseBuilder;
 import static uk.gov.pay.connector.model.api.ExternalChargeRefundAvailability.EXTERNAL_AVAILABLE;
 import static uk.gov.pay.connector.model.domain.ChargeEntityFixture.aValidChargeEntity;
-import static uk.gov.pay.connector.model.domain.ChargeStatus.CAPTURED;
-import static uk.gov.pay.connector.model.domain.ChargeStatus.CREATED;
-import static uk.gov.pay.connector.model.domain.ChargeStatus.ENTERING_CARD_DETAILS;
+import static uk.gov.pay.connector.model.domain.ChargeStatus.*;
```
Comment (georgeracu) [inline src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java]: We do not use wildcard imports :-)

File: src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java [hunk 6]
```diff
@@ -164,37 +143,13 @@ public void shouldCreateACharge() throws Exception {
...
-        assertThat(createdPaymentRequestEntity.getGatewayAccount().getGatewayName(), is("sandbox"));
-        assertThat(createdPaymentRequestEntity.getReference(), is(ServicePaymentReference.of("Pay reference")));
-        assertThat(createdPaymentRequestEntity.getDescription(), is("This is a description"));
-        assertThat(createdPaymentRequestEntity.getAmount(), is(100L));
-        assertThat(createdPaymentRequestEntity.getReturnUrl(), is("http://return-service.com"));
-        assertThat(createdPaymentRequestEntity.getCreatedDate(), is(ZonedDateTimeMatchers.within(3, ChronoUnit.SECONDS, ZonedDateTime.now(ZoneId.of("UTC")))));
-    }
-
     @Test
     public void shouldUpdateEmailToChargeTransaction() {
```
Comment (georgeracu) [inline src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java]: Maybe rename it to `shouldUpdateEmailToCharge`, as we don't have ChargeTransaction anymore

File: src/main/java/uk/gov/pay/connector/service/ChargeStatusUpdater.java [hunk 7]
```diff
@@ -15,44 +13,15 @@
 public class ChargeStatusUpdater {
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/service/ChargeStatusUpdater.java]: This class needs removed. Has empty methods

File: src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java [hunk 8]
```diff
@@ -15,27 +15,17 @@
...
 
 import javax.inject.Inject;
-import javax.ws.rs.GET;
-import javax.ws.rs.HeaderParam;
-import javax.ws.rs.POST;
-import javax.ws.rs.Path;
-import javax.ws.rs.PathParam;
-import javax.ws.rs.Produces;
-import javax.ws.rs.QueryParam;
+import javax.ws.rs.*;
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java]: Wildcard imports

File: src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java [hunk 9]
```diff
@@ -47,12 +37,7 @@
 import static uk.gov.pay.connector.service.ChargeExpiryService.EXPIRABLE_STATUSES;
 import static uk.gov.pay.connector.service.search.SearchService.TYPE.CHARGE;
 import static uk.gov.pay.connector.service.search.SearchService.TYPE.TRANSACTION;
-import static uk.gov.pay.connector.util.ResponseUtil.fieldsInvalidResponse;
-import static uk.gov.pay.connector.util.ResponseUtil.fieldsInvalidSizeResponse;
-import static uk.gov.pay.connector.util.ResponseUtil.fieldsMissingResponse;
-import static uk.gov.pay.connector.util.ResponseUtil.notFoundResponse;
-import static uk.gov.pay.connector.util.ResponseUtil.responseWithChargeNotFound;
-import static uk.gov.pay.connector.util.ResponseUtil.successResponseWithEntity;
+import static uk.gov.pay.connector.util.ResponseUtil.*;
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java]: Wildcard imports

File: src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java [hunk 10]
```diff
@@ -47,12 +37,7 @@
...
 
 @Path("/")
 public class ChargesApiResource {
```
Comment (georgeracu) [inline src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java]: Besides wildcard imports, no other changes here

**Review by georgeracu — CHANGES_REQUESTED** [review]
Overall a big piece of work with a clean removal, only a few comments mainly related to code style and one class that needs to be removed as well.

File: src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java (hunk 1 above)
Comment (kbottla) [reply src/test/java/uk/gov/pay/connector/it/resources/ChargesApiResourceITest.java]: Nothing for now :)

File: src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java (hunk 2 above)
Comment (kbottla) [reply src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java]: Removed wild card imports

File: src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java (hunk 3 above)
Comment (kbottla) [reply src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java:23]: Removed wild card imports

File: src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java (hunk 4 above)
Comment (kbottla) [reply src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java]: Removed wild card imports

File: src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java (hunk 5 above)
Comment (kbottla) [reply src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java]: Removed wild card imports

File: src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java (hunk 6 above)
Comment (kbottla) [reply src/test/java/uk/gov/pay/connector/service/ChargeServiceTest.java]: Sounds good to me, renamed method.

File: src/main/java/uk/gov/pay/connector/service/ChargeStatusUpdater.java (hunk 7 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/service/ChargeStatusUpdater.java]: Removed class ChargeStatusUpdater

File: src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java (hunk 8 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java]: Removed wild card imports

File: src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java (hunk 9 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java]: Removed wild card imports

File: src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java (hunk 10 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java]: Removed wild card imports

**Review by georgeracu — APPROVED** [review]
Looks good 👍

**Pull request merged**